File: dsaWithJava/tree/binarytree/MaxNode.java
package tree.binarytree;

public class MaxNode {
    public static int getMax(Node root) {
        if (root == null)
            return Integer.MIN_VALUE;
        return Math.max(root.data, Math.max(getMax(root.left), getMax(root.right)));
    }

    public static void main(String[] args) {
        int[] nodes = { 1, 2, 4, -1, -1, 5, -1, -1, 3, -1, 6, -1, -1 };
        Node root = TreeBuilder.buildTree(nodes);
        System.out.println(getMax(root));
    }
}

File: dsa_with_python/tree/binarytree/max_node.py
import sys
from node import Node
from tree_builder import build_tree


def get_max(root: Node) -> int:
    if root == None:
        return -sys.maxsize - 1
    return max(root.data, get_max(root.left), get_max(root.right))


# nodes = [1, 2, 4, -1, -1, 5, -1, -1, 3, -1, 6, -1, -1]
# root = build_tree(nodes)
# print(get_max(root))
